gearcmd/worker.go

```diff
@@ -28,42 +30,66 @@ type TaskConfig struct {
 	RetryCount   int
 }
 
+var (
+	lg = logger.New("gearcmd")
+)
+
 // Process runs the Gearman job by running the configured task.
 // We need to implement the Task interface so we return (byte[], error)
 // though the byte[] is always nil.
 func (conf TaskConfig) Process(job baseworker.Job) ([]byte, error) {
+	jobID := getJobID(job)
+	if jobID == "" {
+		jobID = strconv.Itoa(rand.Int())
+		lg.InfoD("rand-job-id", logger.M{"msg": "no job id parsed, random assigned."})
+	}
+
+	// We create a temporary directory to be used as the work directory of the process. Currently
+	// we do not handle the case of this being used multiple times when retrying, files are
+	// expected to be overwritten.
+	tempDirPath, err := ioutil.TempDir("/tmp", fmt.Sprintf("%s-%s", conf.FunctionName, jobID))
```

Comment:
Also, this is a behavior change, but probably for the better: how do you feel about, if it exists, using the `$MESOS_SANDBOX` environment var as the base directory? That will allow us to browse the files created by a job in the mesos UI, making it much easier to debug issues where a worker is leaking disk